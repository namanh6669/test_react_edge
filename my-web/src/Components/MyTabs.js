import React from 'react';
import MyTables from './MyTables.js'
import { Tabs, Tab } from 'material-ui/Tabs';

const styles = {
    bar: {
        width:'30%',

    },
    headline: {

        fontSize: 14,
        paddingTop: 10,
        marginBottom: 5,
        fontWeight: 400,


    },
    inkbar: {
        height: 5,
        width: '50%',
    },

   
};

class MyTabs extends React.Component {

    constructor(props) {
        super(props);
    }
    sendValueIncrease =()=>{
        this.props.callbackTypeSort("Decrease");
    };
    sendValueDecrease =()=>{
        this.props.callbackTypeSort("Increase");
    };

    
    render() {
        return (
            <Tabs
               style ={styles.bar}
                inkBarStyle={styles.inkbar}
                
                
            >
                <Tab  className="gain" style={styles.headline} label=" Top Gainers " value="Increase"  onActive={this.sendValueIncrease}>

                </Tab>
                <Tab  className="lose" style={styles.headline} label=" Top Loser " value="Decrease" onActive={this.sendValueDecrease}>
                </Tab>
            </Tabs>
        );
    }
} export default MyTabs;