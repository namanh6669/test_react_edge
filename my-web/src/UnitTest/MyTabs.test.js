import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import MyTabs from '../Components/MyTabs'

const mytabs = shallow(<MyTabs/>);

test('Render MyTabs',()=>{
    expect(toJson(mytabs)).toMatchSnapshot();
})
test('Top Losers Tab', () => {
    const tabLose = mytabs.find('.lose');
    tabLose.simulate('active');
    expect(mytabs.state().sortType).toEqual('lose')
});  
test('Top Gainner Tab', () => {
    const tabLose = mytabs.find('.gain');
    tabLose.simulate('active');
    expect(mytabs.state().sortType).toEqual('gain')
});