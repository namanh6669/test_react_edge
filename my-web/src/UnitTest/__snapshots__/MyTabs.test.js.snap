// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MyTabs 1`] = `<mytabs />`;
